src/core/aws-net-websocket-registry.adb

```diff
@@ -1409,19 +1364,28 @@ package body AWS.Net.WebSocket.Registry is
       Message      : Unbounded_String;
       Is_Binary    : Boolean := False;
       Timeout      : Duration := Forever;
-      Asynchronous : Boolean := False) is
+      Asynchronous : Boolean := False)
+   is
+      use Ada.Strings.Unbounded.Aux;
+      S  : Big_String_Access;
+      L  : Natural;
    begin
-      DB.Send (Socket, Message, Is_Binary, Timeout, Asynchronous);
+      Get_String (Message, S, L);
+      DB.Send (Socket, S (1 .. L), Is_Binary, Timeout, Asynchronous);
    end Send;
 
    procedure Send
      (Socket       : in out Object'Class;
       Message      : Stream_Element_Array;
       Is_Binary    : Boolean := True;
       Timeout      : Duration := Forever;
-      Asynchronous : Boolean := False) is
+      Asynchronous : Boolean := False)
+   is
+      A : String (Natural (Message'First) .. Natural (Message'Last))
```

Comment:
This may not been working in Character is not the size of a byte. This is way in Translator.Conversion we have a fast version (doing what you've done there) or a slow version doing the conversion.

Reply:
At this stage, I don't think there is a platform supported by GNAT where a character is not a byte. So we could add a pragma Assert just to make sure.
The above is more efficient than AWS.Translator, because To_String returns the result on the secondary stack, which requires extra copies of the string (and definitely has an impact when sending large messages.
Also since To_String needs to store the result string somewhere, it will be on the stack. This fails if the string is larger than 2Mb (with the default stack size for tasks)